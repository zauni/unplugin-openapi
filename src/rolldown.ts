import { unplugin } from "./index.js";

export default unplugin.rolldown as typeof unplugin.rolldown;
