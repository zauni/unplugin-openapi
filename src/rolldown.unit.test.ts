import { rolldown } from "rolldown";
import { describe, expect, it } from "vitest";
import rolldownPlugin from "./rolldown.js";

describe("Rolldown plugin", () => {
	it("should handle a YAML schema with a reference", async () => {
		const build = await rolldown({
			input: "src/fixtures/esbuild/main.ts",
			plugins: [rolldownPlugin()],
		});

		const result = await build.generate();
		const { code } = result.output[0];

		expect(code).toMatchSnapshot();
		expect(code).toContain('properties: { someKey: { type: "string" } },');
	});

	it("should handle a JSON schema with a reference", async () => {
		const build = await rolldown({
			input: "src/fixtures/esbuild/main-json.ts",
			plugins: [
				rolldownPlugin({
					extensions: [".openapi.json"],
				}),
			],
		});

		const result = await build.generate();
		const { code } = result.output[0];

		expect(code).toMatchSnapshot();
		expect(code).toContain('properties: { someKey: { type: "string" } },');
	});
});
