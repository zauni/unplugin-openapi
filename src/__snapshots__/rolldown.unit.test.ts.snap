// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Rolldown plugin > should handle a JSON schema with a reference 1`] = `
"
//#region src/fixtures/api.openapi.json
var api_openapi_default = {
	openapi: "3.0.0",
	info: {
		title: "My great API",
		description: "Great description",
		version: "1.0.0"
	},
	paths: { "/my/path": { get: {
		summary: "Some GET request",
		responses: { "200": {
			description: "Some response",
			content: { "application/json": {
				schema: {
					type: "object",
					properties: { someKey: { type: "string" } },
					required: ["someKey"]
				},
				example: { someKey: "some value" }
			} }
		} }
	} } }
};

//#endregion
//#region src/fixtures/esbuild/main-json.ts
console.log(api_openapi_default);

//#endregion"
`;

exports[`Rolldown plugin > should handle a YAML schema with a reference 1`] = `
"
//#region src/fixtures/api.yaml
var api_default = {
	openapi: "3.0.0",
	info: {
		title: "My great API",
		description: "Great description",
		version: "1.0.0"
	},
	paths: { "/my/path": { get: {
		summary: "Some GET request",
		responses: { "200": {
			description: "Some response",
			content: { "application/json": {
				schema: {
					type: "object",
					properties: { someKey: { type: "string" } },
					required: ["someKey"]
				},
				example: { someKey: "some value" }
			} }
		} }
	} } }
};

//#endregion
//#region src/fixtures/esbuild/main.ts
console.log(api_default);

//#endregion"
`;
